src/components/dashboard/burn_tokens.tsx

```diff
@@ -310,58 +294,41 @@ export default function BurnTokens(props: mintProps) {
 			return
 		}
 
-		setAmount(parseUnits(parsedAmount, inverseTokenDecimals.toNumber()))
+		setAmount(parseUnitsBnJs(parsedAmount, inverseTokenDecimals.toNumber()))
 
-		const decimaledParsedAmount = parseUnits(parsedAmount,
+		const decimaledParsedAmount = parseUnitsBnJs(parsedAmount,
 			inverseTokenDecimals.toNumber()
 		)
 		const reserveAmount = BigNumber.from(bondingCurveParams.reserveAmount)
 		const inverseTokenSupply = BigNumber.from(bondingCurveParams.inverseTokenSupply)
 
 		// this should be a non-under/overflow number between 0,1
-		const fee = parseUnits(
+		const fee = parseUnitsBnJs(
 			Number(
 				Number(formatUnits(decimaledParsedAmount, inverseTokenDecimals)) *
 					totalFeePercent
 			).toFixed(Number(inverseTokenDecimals)),
-			inverseTokenDecimals
+			inverseTokenDecimals.toNumber()
 		)
 		const burnedAmount = decimaledParsedAmount.sub(fee)
-		const supplyDelta =
-			Number(formatUnits(inverseTokenSupply.sub(burnedAmount), inverseTokenDecimals)) /
-			Number(formatUnits(inverseTokenSupply, inverseTokenDecimals))
-
-		// this will be a negative number
-		const logSupplyDeltaTimesUtilization =
-			Math.log(supplyDelta) * curveUtilization
 
-		const liquidityReceived = parseUnits(
-			Number(
-				-1 *
-					(Math.exp(logSupplyDeltaTimesUtilization) - 1) *
-					Number(formatUnits(reserveAmount, defaultDecimals))
-			).toFixed(defaultDecimals)
-			,
-			defaultDecimals
-		)
+		const supplyDeltaBig = bignumber(inverseTokenSupply.sub(burnedAmount).toString()).dividedBy(bignumber(inverseTokenSupply.toString()))
+		const liquidityReceivedBig = reserveAmount.sub(mulPercent(reserveAmount, supplyDeltaBig.sqrt().toNumber()))
 
 		// calculate spot price post mint
-		const curveInvariant = Number(formatUnits(reserveAmount, defaultDecimals)) / Math.pow(Number(formatUnits(inverseTokenSupply, inverseTokenDecimals)), curveUtilization) 
-		const newPrice = curveInvariant * curveUtilization
-		/
-		Math.pow(Number(formatUnits(inverseTokenSupply.sub(burnedAmount), inverseTokenDecimals)), 1 - curveUtilization)
+		const curveInvariantBig = BigNumber.from(bondingCurveParams.invariant)
+		const newPriceBig = curveInvariantBig.mul(parseUnits(curveUtilization.toString(), defaultDecimals)).div(computeSquareRoot(inverseTokenSupply.sub(burnedAmount))).div(parseUnits("1", defaultDecimals/2))
```

Comment:
From contract side, price is calculated by: utilization * reserve / ibcSupply

Reply:
ah thanks, this is way simpler than the formula on the docs

Reply:
fixed in https://github.com/bear-market-labs/ibc_interface/pull/252/commits/655f61def832c9e947bb5e1de05afe131bb8268f